api/internal/handler/nft_handler.go

```diff
@@ -70,3 +80,133 @@ func NewUpdateNftBackstoryHandler(nftTokenRepository repository.NftTokenReposito
 		c.JSON(http.StatusOK, nftToken)
 	}
 }
+
+func NewGetNftsHandler(nftTokenRepository repository.NftTokenRepository) func(c *gin.Context) {
+	return func(c *gin.Context) {
+
+	}
+}
+
+func NewUpdateNftHandler(nftTokenRepository repository.NftTokenRepository, client *ethclient.Client) func(c *gin.Context) {
+	return func(c *gin.Context) {
+		franchiseId, err := strconv.ParseInt(c.DefaultQuery("franchiseId", ""), 10, 64)
+		if err != nil {
+			logger.Errorf("Failed to convert franchise id: %v", err)
+			c.String(http.StatusBadRequest, "franchise id is invalid")
+			return
+		}
+
+		tokenId, err := strconv.ParseInt(c.Param("id"), 10, 64)
+		if err != nil {
+			logger.Errorf("Failed to convert token id: %v", err)
+			c.String(http.StatusBadRequest, "token id is invalid")
+			return
+		}
+
+		collectionAddress := c.DefaultQuery("collectionAddress", "")
+		if collectionAddress == "" {
+			c.String(http.StatusBadRequest, fmt.Sprintf("input address is invalid, address: %s", collectionAddress))
+			return
+		}
+
+		address := common.HexToAddress(collectionAddress)
+		contract, err := abi.NewAbi(address, client)
+		if err != nil {
+			logger.Errorf("Failed to instantiate the contract: %v", err)
+			c.String(http.StatusInternalServerError, "Internal server error")
+			return
+		}
+
+		uri, err := contract.TokenURI(nil, big.NewInt(tokenId))
+		if err != nil {
+			logger.Errorf("Failed to query uri: %v", err)
+			c.String(http.StatusInternalServerError, "Internal server error")
+			return
+		}
+
+		var nft *repository.NftTokenModel
+		if strings.Contains(uri, "ipfs") {
+			nft, err = createIpfsNftRecord(uri, franchiseId, int(tokenId), collectionAddress)
+			logger.Errorf("Failed to construct ipfs nft record: %v", err)
+		} else {
```

Comment:
Seems we only support IPFS or on-chain SVG?

Reply:
Kingter just implemented and sent me a unified version of the interface. I will use that instead of separate handling of different format